### PR TITLE
Feat/tsemach/support create app from template

### PR DESCRIPTION
**Pull request by tsemachLi**
https://monday.monday.com/boards/3670992828/pulses/5945078169

### CODE REVIEW

File: src/services/git-service.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        if (code === 0) return resolve();
+        reject(new Error(`Failed to clone ${gitRepoUrl}`));
+      });
+      cloneProcess.stderr?.on('data', data => {
+        output(prettyPrint(data as string));
+      });
+    });
+
+    // Move the specific folder to the desired location
+    await fs.move(path.join(tempRepoPath, folderPath), targetPath, { overwrite: true });
```
Comment (DorShakedMonday) [inline src/services/git-service.ts:44]: Try using the native nodejs `fs` capabilities for this

File: src/commands/app/__tests__/create.test.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+const requestSpy = getRequestSpy();
+
+describe('app:create', () => {
+  jest.setTimeout(1_000_000);
+  const MOCK_BASE_RESPONSE = { headers: {} };
+  const MOCK_APP_ID = 1_000_000_001;
+  const MOCK_APP_VERSION_ID = 1_000_000_002;
+  const MOCK_APP_FEATURE_ID = 1_000_000_003;
+  const question = 'Select a template to start with';
+  const answer = 'Fullstack React + Node.js (TypeScript)';
```
Comment (DorShakedMonday) [inline src/commands/app/__tests__/create.test.ts:21]: This should be a part of a specific `it` no? Every test case in here will go with the same set of `question` & `answer`

File: src/commands/app/__tests__/create.test.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    },
+  };
+
+  const mockTunnelingDomainResponse = {
+    domain: 'abcdefgh.monday.mock',
+  };
+
+  it('should create app successfully', async () => {
+    const mockPushFlags = buildMockFlags(AppCreate, { name: 'New App by CLI' });
+    mockListPromptImplementation([{ question, answer }]);
```
Comment (DorShakedMonday) [inline src/commands/app/__tests__/create.test.ts:68]: Where do you reset this mock?

File: src/commands/app/__tests__/create.test.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        return { data: mockAppFeatureBuildResponse, ...MOCK_BASE_RESPONSE };
+      }
+    });
+    await AppCreate.run(mockPushFlags);
+    const stdout = getStdout();
+    expect(stdout).toContain('Your app is ready');
+    expect(stdout).toContain("cd ./quickstart-fullstack-react-node' to see your app files.");
+    expect(stdout).toContain(
+      `open in browser: http://monday.llama.fan/apps/manage/${MOCK_APP_ID}/app_versions/${MOCK_APP_VERSION_ID}/sections/appDetails to manage your app`,
+    );
```
Comment (DorShakedMonday) [inline src/commands/app/__tests__/create.test.ts:110]: This is the only interesting check in here, no?

**Review by DorShakedMonday — APPROVED** [review]

File: src/services/git-service.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { spawn } from 'node:child_process';
+import * as path from 'node:path';
+
+import fs from 'fs-extra';
```
[the remark this answers is not in the document]
Comment (tsemachLi) [reply src/services/git-service.ts:4]: much more easy to use for moving and removing folder

**Pull request merged**